substrate/frame/utility/src/lib.rs

```diff
@@ -454,6 +463,98 @@ pub mod pallet {
 			let res = call.dispatch_bypass_filter(frame_system::RawOrigin::Root.into());
 			res.map(|_| ()).map_err(|e| e.error)
 		}
+
+		/// Dispatch a fallback call in the event the main call fails to execute.
+		///
+		/// This function first attempts to dispatch the `main` call. If the `main` call fails, the `fallback` call
+		/// is dispatched instead. Both calls are executed with the same origin, and the weight of both calls
+		/// is accumulated and returned. The success or failure of the main and fallback calls is tracked and
+		/// appropriate events are deposited.
+		///
+		/// May be called from any origin except `None`.
+		///
+		/// - `main`: The main call to be dispatched. This is the primary action to execute.
+		/// - `fallback`: The fallback call to be dispatched in case the `main` call fails.
+		///
+		/// ## Dispatch Logic
+		/// - If the origin is `root`, both the main and fallback calls are executed without applying any origin filters.
+		/// - If the origin is not `root`, the origin filter is applied to both the `main` and `fallback` calls.
+		///
+		/// ## Complexity
+		/// - O(1) for the origin check and dispatching the main call.
+		/// - O(1) for the origin check and dispatching the fallback call (if needed).
+		/// - Overall complexity is O(1) since we only dispatch at most two calls.
+		///
+		/// ## Weight
+		/// The weight of this call is calculated as the sum of:
+		/// - The weight of the `main` call (if it is executed),
+		/// - The weight of the `fallback` call (if the `main` call fails and the `fallback` is executed),
+		/// - A base weight (`WeightInfo::if_else()`), which accounts for the logic involved in dispatching and handling both calls.
+		#[pallet::call_index(6)]
+		#[pallet::weight({
+			let main_dispatch_info = main.get_dispatch_info();
+			let fallback_dispatch_info = fallback.get_dispatch_info();
+			(
+				T::WeightInfo::if_else()
+					.saturating_add(main_dispatch_info.call_weight)
+					.saturating_add(fallback_dispatch_info.call_weight),
+				main_dispatch_info.class,
```

Comment:
I think we should take the lowest class between main and fallback.

Otherwise somebody can do a call with a failing operational main call. And a fallback with a batch of normal call.

Reply:
We don't need to handle `Mandatory` class, same as batch.
So we can just do: `if main.class == Operational && fallback.class == Operational { Operational } else { Normal }